cql3/statements/create_view_statement.cc

```diff
@@ -141,6 +143,14 @@ std::pair<view_ptr, cql3::cql_warnings_vec> create_view_statement::prepare_view(
         throw exceptions::invalid_request_exception(format("Cannot create a materialized view on a table in a separate keyspace ('{}' != '{}')",
                 _base_name.get_keyspace(), keyspace()));
     }
+    const sstring& cf_name = column_family();
+    boost::regex name_regex("\\w+");
+    if (!boost::regex_match(std::string(cf_name), name_regex)) {
+        throw exceptions::invalid_request_exception(format("\"{}\" is not a valid materialized view name (must be alphanumeric character only: [0-9A-Za-z]+)", cf_name.c_str()));
```

Comment:
I guess you copied this message verbatim from create_table_statement.cc, but it is actually incorrect, the "\w" in Boost (see https://www.boost.org/doc/libs/1_33_1/libs/regex/doc/character_class_names.html but the same applies to other regex implementations as well) actually allows alphanumerics **and underscores**. I.e., the regular expression in the message should be `[0-9A-Za-z_]+` (note I added an underscore).
Also the "+" made me think - where do we actually prevent creating a table with an **empty** name? We don't check if size()==0 here, but is it checked somewhere else?

Reply:
@nyh I've added a test for size() == 0. There is a syntax error.
The message also is fixed.